OpenSim/Analyses/OutputReporter.h

```diff
@@ -0,0 +1,102 @@
+#ifndef OUTPUT_REPORTER_H_
+#define OUTPUT_REPORTER_H_
+/* -------------------------------------------------------------------------- *
+ *                         OpenSim: OutputReporter.h                          *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2017 Stanford University and the Authors                *
+ * Author(s):                                                                 *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+
+//=============================================================================
+// INCLUDES
+//=============================================================================
+#include <OpenSim/Simulation/Model/Analysis.h>
+#include <OpenSim/Common/Reporter.h>
+#include <OpenSim/Common/STOFileAdapter.h>
+#include "osimAnalysesDLL.h"
+
+#ifdef SWIG
+    #ifdef OSIMANALYSES_API
+        #undef OSIMANALYSES_API
+        #define OSIMANALYSES_API
+    #endif
+#endif
+//=============================================================================
+//=============================================================================
+namespace OpenSim { 
+
+/**
+ * The OutputReporter Analysis is a wrapper for a TableReporter. It generates
+ * and writes TimeSeriesTables of Output values according to the names listed
+ * as properties of the Analysis. OutputReporter enables the AnalyzeTool to
+ * report on Outputs via the Analysis interface but backed by a TableReporter.
+ * The OutputReporter currently only supports Outputs of type, double, Vec3
+ * and SpatialVec.
```

Comment:
I think we should either rename this class to `OutputAnalysis` (which would be in line with the naming of `MuscleAnalysis`), or we should add a comment here saying "@note This class does *not* derive from Reporter.". I *strongly* prefer the first option. I am fairly certain users will be confused if we name this class  OutputReporter.

Reply:
Should we then rename `ForceReporter` and `ProbeReporter`? I can rename it to `OutputAnalysis` but there is no *analysis* being performed - we just report outputs.  I agree it can be confusing if you are trying to understand the class hierarchy. But, as a user of a `Tool`, I think `OutputReporter` is clearer and corresponds to the Force and Probe reporters.

Reply:
> Should we then rename ForceReporter and ProbeReporter?

I wouldn't mind this, other than not wanting to rename existing classes. Alternatively, if you leave this class named as OutputReporter, shouldn't you rename MuscleAnalysis to MuscleReporter?

If a user has a CMC simulation and they want to know joint reactions during that simulation, I don't think they would be confused by the language "analyze your CMC simulation to obtain joint reactions." I'm not sure it matters to the user that the calculation of the joint reactions actually happens in the Model, not in the Analysis class.

Perhaps there's an alternative name that can please everyone. OutputAggregator, OutputCollector.